core/platform/email/gae_email_services_test.py

```diff
@@ -36,10 +45,170 @@ def test_sending_email(self):
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(0, len(messages))
 
+        messages = self.mail_stub.get_sent_messages(
+            to=feconf.ADMIN_EMAIL_ADDRESS)
+        self.assertEqual(0, len(messages))
         with self.swap(feconf, 'CAN_SEND_EMAILS', True):
             gae_email_services.send_mail(
-                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                feconf.SYSTEM_EMAIL_ADDRESS,
+                feconf.ADMIN_EMAIL_ADDRESS,
                 'subject', 'body', 'html', bcc_admin=False)
         messages = self.mail_stub.get_sent_messages(
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(1, len(messages))
+
+    def test_that_email_bcc_is_sent_if_bcc_admin_is_true(self):
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=True)
+        messages = self.mail_stub.get_sent_messages(
+            to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertEqual(messages[0].bcc, feconf.ADMIN_EMAIL_ADDRESS)
+
+    def test_that_email_bcc_not_sent_if_bcc_admin_is_false(self):
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
```

Comment:
Follow this comment:

> Before the action you should check that get_sent_messages() returns 0.
> 
> Ditto below. See the guidelines for writing backend tests.

Reply:
@DubeySandeep @seanlip 
Sorry for marking it as resolved, that was a mistake. Actually, I wanted to ask something about it.
I am not able to wrap my head around as to why I need to do that.
Though I have put up code for doing this in the latest commit.
It would be great if you could help me out here.
Thanks!

Reply:
@siddhantkhandelwal, We use to follow a pattern for writing backend tests (see [wiki page](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests) for more info):

> 1. Tests should use the following general pattern:
> 
> (i) setup - this is where you prepare any inputs/environment needed for the test.
> (ii) baseline verification - check the values without performing any action.
> (iii) action - perform the action or function call that leads to the expected change.
> (iv) endline verification - check that the values in the baseline verification have changed accordingly.


having a (ii) baseline validation assures that the changes are made only via the (iii) action step.

Reply:
@siddhantkhandelwal, have you followed this comment, everywhere possible? (As asked by Sean)

I cannot see any such check inside `test_sending_email_with_reply_to_id_adds_reply_to_email` etc.

Please reply to the comment when you completely address them.

Reply:
@DubeySandeep PTAL!
I have added Baseline Verification to the tests.
Thanks!

Reply:
@dubeysandeep
PTAL

Reply:
@DubeySandeep @seanlip 
PTAL!